jpsreport/scripts/helpers/Ploter.py

```diff
@@ -0,0 +1,167 @@
+    print("---> ", figname)
+
+def plot_profiles(Id, field_dir, geo_filename, beginsteady=None, endsteady=None):
+    """ Plot density, velocity and flow
+
+    Note: J = rho*v
+
+    :param Id: measurement area
+    :param field_dir:
+    :param beginsteady: From frame. If None, then take first frame
+    :param endsteady: End frame. If None, then take last frame
+    :param geo_filename: geometry file as parsed from the inifile
+    :returns: png-plot
+    :rtype:
+
+    """
+
+        # for j in range(beginsteady, endsteady):
+        # density_file = os.path.join(density_files_path, "Prf_d_%s_id_1_%.5d.dat" %(nametraj, j))
+
+    density_files = os.path.join(field_dir, "density",  "Prf_*id_{}_*".format(Id))
+    velocity_files = os.path.join(field_dir, "velocity",  "Prf_*id_{}_*".format(Id))
+    v_Voronoi = glob.glob(velocity_files)
+    f_Voronoi = glob.glob(density_files)
+    # get frames
+    frames = []
+    for f in v_Voronoi:
+        frame = int(f.split("_")[-1].split(".")[0]) #Prf_d_*_id_%d_%.5d.dat)
+        frames.append(frame)
+
+    frames = sorted(frames)
+    # correct arguments if necessary
+    if beginsteady is None:
+        beginsteady = frames[0]
+    elif beginsteady < frames[0]:
+        beginsteady = frames[0]
+
+    if endsteady is None:
+        endsteady = frames[-1]
+    elif endsteady > frames[-1]:
+        endsteady = frames[-1]
+
+    beginsteady -= frames[0]
+    endsteady -= frames[0]
+
+    # get the shape of the matrices
+    shape = np.shape(np.loadtxt(f_Voronoi[0]))
+    density = np.zeros(shape)
+    velocity = np.zeros(shape)
+    xml_datei = open(geo_filename, "r")
+    geo_xml = parse(xml_datei)
+    xml_datei.close()
+    geominX, geomaxX, geominY, geomaxY = geo_limits(geo_xml)
+
+    geometry_wall = read_subroom_walls(geo_xml)
+    geometry_obst = read_obstacle(geo_xml)
+#  -------- density
+    for density_file in f_Voronoi[beginsteady:endsteady+1]:
+        if os.path.exists(density_file):
+            density += np.loadtxt(density_file)
+
+    density = density / (endsteady-beginsteady)
+#  --------- velocity
+    for velocity_file in v_Voronoi[beginsteady:endsteady+1]:
+        if os.path.exists(velocity_file): 
+            velocity += np.loadtxt(velocity_file)
+
+    velocity = velocity / (endsteady-beginsteady)
+    flow = density * velocity
+   
+ # plot
+    figs, axs = plt.subplots(3, 1)
+    axs[0].set_aspect('equal')
+
+    for g in geometry_obst.keys():
+        axs[0].add_patch(ppolygon(geometry_obst[g], color='white'))
+        axs[1].add_patch(ppolygon(geometry_obst[g], color='white'))
+        axs[2].add_patch(ppolygon(geometry_obst[g], color='white'))
+
+    for gw in geometry_wall.keys():
+        axs[0].plot(geometry_wall[gw][:, 0], geometry_wall[gw][:, 1], color='white', lw=1)
+        axs[1].plot(geometry_wall[gw][:, 0], geometry_wall[gw][:, 1], color='white', lw=1)
+        axs[2].plot(geometry_wall[gw][:, 0], geometry_wall[gw][:, 1], color='white', lw=1)
```

Comment:
@gjaeger Here, the geometry is plotted along with the profiles. Is there another scenario, where the geometry should be plotted as well? Is there a need to plot *just* the geometry?

Reply:
A general idea: The plot function for the geometry only can be used for presentation and explanation. These plots could be supplemented by the illustration of the measurement areas. You could use the information from the ini-file for jpsreport. This could save some time.